src/main/java/uk/gov/hmcts/cmc/claimstore/documents/content/StatementOfMeansContentProvider.java

```diff
@@ -0,0 +1,92 @@
+package uk.gov.hmcts.cmc.claimstore.documents.content;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.domain.models.statementofmeans.BankAccount;
+import uk.gov.hmcts.cmc.domain.models.statementofmeans.Child;
+import uk.gov.hmcts.cmc.domain.models.statementofmeans.CourtOrder;
+import uk.gov.hmcts.cmc.domain.models.statementofmeans.Debt;
+import uk.gov.hmcts.cmc.domain.models.statementofmeans.Dependant;
+import uk.gov.hmcts.cmc.domain.models.statementofmeans.Employment;
+import uk.gov.hmcts.cmc.domain.models.statementofmeans.Expense;
+import uk.gov.hmcts.cmc.domain.models.statementofmeans.Income;
+import uk.gov.hmcts.cmc.domain.models.statementofmeans.OnTaxPayments;
+import uk.gov.hmcts.cmc.domain.models.statementofmeans.OtherDependants;
+import uk.gov.hmcts.cmc.domain.models.statementofmeans.Residence;
+import uk.gov.hmcts.cmc.domain.models.statementofmeans.SelfEmployment;
+import uk.gov.hmcts.cmc.domain.models.statementofmeans.StatementOfMeans;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+@Component
+public class StatementOfMeansContentProvider {
+
+    public Map<String, Object> createContent(StatementOfMeans statementOfMeans) {
```

Comment:
consider extracting each part of statement of means to its own method, 
this method is very long with lots of nested if statements, I wouldn't be surprised if sonar complains about the complexity of it

Reply:
done